src/BTVNanoCommissioning/workflows/ctag_DY_valid_sf.py

```diff
@@ -143,6 +157,15 @@ def process(self, events):
         sz = sposmu + snegmu
         sjets = event_jet[event_level]
         njet = ak.count(sjets.pt, axis=1)
+        # Find the PFCands associate with selected jets. Search from jetindex->JetPFCands->PFCand
+        if self._campaign == "Winter22Run3":
```

Comment:
Not all the PFCands were there for run2 samples

Reply:
From a PFNano perspective, there is nothing stopping you from applying this linking between Jets / JetPFCands / PFCands also with Run2 samples that you can do with this newest campaign, as everything got backported or was already there. The thing I can see which is new for Run3 in this particular context (and should be masked for Run2) is related to new variables [`numberOFHits()`, `numberOfPixelHits()`], not the usage of all / not-all candidates themselves. For `lostInnerHits()`, a protection is not necessary, as it is also available for 106X etc. in Run2 PFNano samples.

Reply:
> same comment as above, we need to keep in mind to add future releases there or to remove Run 2 support at some point

@demuller  Indeed I should rewrite the protection where self._campaign != "Rereco17_94X" which is a very old release (PFNanov2) so the newer version will also store these variables.



> From a PFNano perspective, there is nothing stopping you from applying this linking between Jets / JetPFCands / PFCands also with Run2 samples that you can do with this newest campaign, as everything got backported or was already there. The thing I can see which is new for Run3 in this particular context (and should be masked for Run2) is related to new variables [`numberOFHits()`, `numberOfPixelHits()`], not the usage of all / not-all candidates themselves. For `lostInnerHits()`, a protection is not necessary, as it is also available for 106X etc. in Run2 PFNano samples.

@AnnikaStein Indeed we can do so if we want other info to be stored in the commissioning as what has been done right now.  The protection there is to avoid the missing variables being used in the selection. But not sure whether we still need this for run2 study as the commissioning has been done already